save/src/protos.proto

```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
+
+package protos;
+
+message Meta {
+    // Number of voxels along the edge of a chunk
+    uint32 chunk_size = 1;
+}
+
+message Node {
+    // Entities whose origins lie within this node
+    repeated Archetype archetypes = 1;
+
+    // Voxel data for each modified chunk
+    repeated Chunk chunks = 2;
+}
+
+// A set of entities, all of which have the same components
+message Archetype {
+    // Entity IDs
+    repeated fixed64 entities = 1;
+
+    // Type of components stored in each column
+    repeated ComponentType component_types = 2;
+
+    // Columns of component data
+    repeated bytes component_data = 3;
```

Comment:
> Should it have the same number of elements as component_types?

Yep. A separate `Column` message would make this a bit clearer, but add extra indirection to the binary encoding.

> are a given component of each entry put together and compressed into a byte array?

Not separately compressed; that'd be redundant, since the whole `Node` gets compressed anyway. Just densely packed according to some component-specific serialization rule (e.g. dense `f32`s if `POSITION` ends up corresponding to a matrix).

Reply:
Reworded comment for clarity.

Reply:
Do we want to edit the test case so that `component_types` and `component_data` have the same number of elements? Based on the reworded comment, it looks like the test case has an archetype with invalid data.

Reply:
These structures have no semantic import as far as this crate is concerned, so I don't think this is necessary; it will not impact the behavior the test is exercising. Unless it's sufficiently confusing as to compromise readability?

Reply:
Right. I think readability is the main potential concern. When I left my latest comment, I believed there was a small chance that I misunderstood, and there were actually cases where the two lists should have a different number of elements. Since unit tests can often serve as a form of documentation, I was treating it as such.

However, I also agree that complicating the test data in a way that doesn't actually affect execution in any way is not ideal, so leaving a comment in the test noting that "`component_data` is left empty artificially for test simplicity despite it being against the spec" would likely avoid this confusion.

Reply:
Easier to fill it in.